examples/ipv4.rs

```diff
@@ -0,0 +1,135 @@
+    let mut filling_ring_size: u32 = 64;
+    let mut completion_ring_size: u32 = 64;
+
+    for i in 1..args.len() {
+        match i {
+            1 => {
+                if_name = args[1].clone();
+            }
+            2 => {
+                chunk_size = args[2].parse::<u32>().expect("Check chunk size.");
+            }
+            3 => {
+                chunk_count = args[3].parse::<u32>().expect("Check chunk count.");
+            }
+            4 => {
+                rx_ring_size = args[4].parse::<u32>().expect("Check rx ring size.");
+            }
+            5 => {
+                tx_ring_size = args[5].parse::<u32>().expect("Check tx ring size.");
+            }
+            6 => {
+                filling_ring_size = args[6].parse::<u32>().expect("Check filling ring size.");
+            }
+            7 => {
+                completion_ring_size = args[7].parse::<u32>().expect("Check filling ring size.");
+            }
+            _ => {
+                panic!("abnormal index");
+            }
+        }
+    }
+
+    /* signal define to end the application */
+    let term: Arc<AtomicBool> = Arc::new(AtomicBool::new(false));
+    let result_sigint: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGINT, Arc::clone(&term));
+    let result_sigterm: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGTERM, Arc::clone(&term));
+    let result_sigabrt: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGABRT, Arc::clone(&term));
+
+    if result_sigint.or(result_sigterm).or(result_sigabrt).is_err() {
+        panic!("signal is forbidden");
+    }
+
+    /* execute pv_open() */
+    let pv_open_option: Option<PvNic> = pv::pv::pv_open(
+        &if_name,
+        chunk_size,
+        chunk_count,
+        rx_ring_size,
+        tx_ring_size,
+        filling_ring_size,
+        completion_ring_size,
+    );
+    let mut nic: PvNic;
+    if let Some(a) = pv_open_option {
+        nic = a;
+    } else {
+        panic!("nic allocation failed!");
+    }
+
+    /* initialize rx_batch_size and packet metadata */
+    let rx_batch_size: u32 = 64;
+    let mut packets: Vec<PvPacket> = Vec::with_capacity(rx_batch_size as usize);
+
+    while !term.load(Ordering::Relaxed) {
+        let received: u32 = pv_receive(&mut nic, &mut packets, rx_batch_size);
+
+        if received > 0 {
+            unsafe {
+                let payload_ptr = packets[0 as usize]
+                    .buffer
+                    .add(packets[0 as usize].start as usize)
+                    .cast_const();
+                if std::ptr::read(payload_ptr.offset(12)) == 0x08
+                    && std::ptr::read(payload_ptr.offset(13)) == 0x00
+                    && std::ptr::read(payload_ptr.offset(14)) >> 4 == 4
+                {
+                    packet_dump(&mut packets[0 as usize]);
+                }
+            }
+        }
```

Comment:
From these lines, `payload_ptr` always has `packets[0]`. However, many packets can be received at once (see line 92). Assuming that two packets are received and these code will always dump the first packet alone.
Try to change these lines to dump packets in a row, not just packet in packets[0].